fairlearn/metrics/_plotter.py

```diff
@@ -0,0 +1,281 @@
+
+    errors : str or list of str
+        The name of the error metrics to plot.
+        Should match columns from the given :class:`fairlearn.metrics.MetricFrame`.
+        Errors quantify the bounds relative to the base metric.
+        Errors must be non-negative.
+
+        Example:
+            If the error for a certain column is :code:`0.01`
+            and the metric is 0.6,
+            then the plotted bounds will be :code:`[0.59, 0.61]`
+
+        Note:
+            The return of the error function should be an scalar of the
+            symmetric errors. e.g. :code:`0.01`
+
+    conf_intervals : str or list of str
+        The name of the confidence intervals to plot.
+        Should match columns from the given :class:`fairlearn.metrics.MetricFrame`.
+
+        Note:
+            The return of the error function should be an array of the lower
+            and upper bounds. e.g. :code:`[0.59, 0.62]`
+
+    subplots : bool, default=True
+        Whether or not to plot metrics on separate subplots
+
+    plot_error_labels : bool, default=False
+        Whether or not to plot numerical labels for the error bounds
+
+    error_labels_precision : int, default=4
+        The number of digits of precision to show for error labels
+
+    error_labels_fontsize : int, default=8
+        The font size to use for error labels
+
+    error_labels_color : str, default="black"
+        The font color to use for error labels
+
+    error_labels_ha : str, default="center"
+        The horizontal alignment modifier to use for error labels
+
+    Returns
+    -------
+    :class:`matplotlib.axes.Axes` or :class:`numpy.ndarray` of them
+    """
+    _check_for_ambiguous_error_metric(errors, conf_intervals)
+    _check_for_valid_metrics_format(metrics)
+
+    metrics = [metrics] if isinstance(metrics, str) else metrics
+    conf_intervals = [conf_intervals] if isinstance(conf_intervals, str) else conf_intervals
+    errors = [errors] if isinstance(errors, str) else errors
+
+    df = metric_frame.by_group
+
+    # only plot metrics that aren't arrays (filters out metric errors)
+    if metrics is None:
+        metrics = []
+        for metric in list(df):
+            if not _is_arraylike(df[metric][0]):
+                metrics.append(metric)
+
+    _check_if_metrics_and_error_metrics_same_length(metrics, errors, conf_intervals)
+    _check_if_metrics_length_zero(metrics)
+
+    # plotting without errors or confidence intervals
+    # Note: Returns early
+    if errors is None and conf_intervals is None:
+        axs = _plot_df(df, metrics, kind, subplots, legend_label, **kwargs)
+        return axs
+
+    df_all_errors = pd.DataFrame([])
+    df_all_bounds = pd.DataFrame([])
+    # plotting with confidence intervals:
+    if conf_intervals is not None:
+        for metric, conf_interval in zip(metrics, conf_intervals):
+            _check_valid_conf_interval(df[conf_interval])
```

Comment:
all validations an be done before getting here

Reply:
are you suggesting to move this validation up earlier in this function? Probably before the for loop to save computation?

Reply:
yes, and to fail before we do any computation if the input is not valid